src/DefaultBuilder/samples/SampleApp/Program.cs

```diff
@@ -14,17 +15,14 @@ namespace SampleApp
 {
     public class Program
     {
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            CreateHostBuilder(args).Build().Run();
-        }
+            await using var webApp = WebApplication.Create(args);
 
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.UseStartup<Startup>();
-                });
+            webApp.MapGet("/", (Func<string>)(() => "Hello, World!"));
```

Comment:
Is it necessary to cast to `Func<string>`? Need some syntactic sugar here I guess because the cast looks weird! 😅

Reply:
It's necessary for now, but we plan to fix it for C# 10. https://github.com/dotnet/csharplang/blob/main/proposals/lambda-improvements.md#implicit-conversion-to-systemdelegate

Reply:
Yes, I thought the same earlier! 😇